src/pymor/reductors/mt.py

```diff
@@ -0,0 +1,169 @@
+            Optional dict with more options for the samdp algorithm.
+        which
+            A string specifying which `k` eigenvalues and eigenvectors to
+            compute when using the eig decomposition, default is "LR":
+
+            - `'SM'`: select eigenvalues with smallest magnitude
+            - `'LR'`: select eigenvalues with largest real part
+            - `'NR'`: select eigenvalues with largest norm(residual) / abs(Re(pole))
+            - `'NS'`: select eigenvalues with largest norm(residual) / abs(pole)
+            - `'NM'`: select eigenvalues with largest norm(residual)
+             
+        Returns
+        -------
+        rom
+            Reduced-order model.
+        """
+        assert 0 < r < self.fom.order
+        assert projection in ('orth', 'biorth')
+        assert decomposition in ('eig', 'samdp')
+        assert method_options is None or isinstance(method_options, dict)
+        if not method_options:
+            method_options = {}
+
+        self.V = self.fom.B.as_range_array().empty(reserve=r)
+        self.W = self.fom.C.as_source_array().empty(reserve=r)
+
+        if decomposition == 'eig':
+            if self.fom.A.sparse:
+                self.logger.warning('Converting operator A to a NumPy array.')
+            A = to_matrix(self.fom.A, format='dense')
+            if not isinstance(self.fom.E, IdentityOperator):
+                if self.fom.E.sparse:
+                    self.logger.warning('Converting operator E to a NumPy array.')
+            E = None if isinstance(self.fom.E, IdentityOperator) else to_matrix(self.fom.E, format='dense')
+
+            if symmetric:
+                poles, ev_r = spla.eig(A, E, right=True)
+                rev = self.fom.A.source.from_numpy(ev_r.T)                
+                lev = rev.copy()
+            else:
+                poles, ev_l, ev_r = spla.eig(A, E, left=True, right=True)
+                rev = self.fom.A.source.from_numpy(ev_r.T)
+                lev = self.fom.A.source.from_numpy(ev_l.T)
+            if which == 'SM':
+                idx = np.argsort(np.abs(poles))
+            elif which == 'LR':
+                idx = np.argsort(-poles.real)
+            else:
+                absres = np.empty(len(poles))
+                for i in range(len(poles)):
+                    b_norm = spla.norm(lev[i].inner(self.fom.B.as_range_array()), ord=2)
+                    c_norm = spla.norm(self.fom.C.as_source_array().inner(rev[i]), ord=2)
+                    absres[i] = b_norm * c_norm
+                if which == 'NR':
+                    idx = np.argsort(-absres / np.abs(np.real(poles)))
+                elif which == 'NS':
+                    idx = np.argsort(-absres / np.abs(poles))
+                elif which == 'NM':
+                    idx = np.argsort(-absres)
+            poles = poles[idx]
+            rev = rev[idx]
+            lev = lev[idx]
+            poles = poles[:r]
+            rev = rev[:r]
+            lev = lev[:r]
```

Comment:
I'm afraid weird things might happen if `poles[:r]` is not closed under conjugation (i.e., if there is a complex eigenvalue, but not also it's complex conjugate). Maybe you can try testing it with an `LTIModel` that has `-1`, `-2 + 1j`, `-2 - 1j`, `-3` as poles and residues all equal, and see what happens when you try reducing it to order 2. I'm actually not sure which poles to pick in such a case. At least in some cases, the ROM order should be exactly `r` (e.g., if modal truncation would be used inside SOR-IRKA instead of IRKA), so completing `poles[:r]` under conjugation would not be good...

Reply:
Yes, this is actually a problem I already thought about when comparing eig and samdp decomposition. The samdp algorithm returns the first r dominant poles closed under conjunction such that the rom would be of higher order than r for complex conjugated pairs of poles. So but as you said it is more important that the ROM order is exactly r? 
Then I will do some research which poles to pick in the case described above.

Reply:
Ah, right. Then it might be strange for the eig and samdp versions to behave differently. Although, SOR-IRKA would only use the dense version.
Maybe @lbalicki has some ideas.

Reply:
In my opinion with either option the order of the ROM should be `r`. I'm not aware of any experiments/results which indicate if it is OK to use only one eigenvector of a complex conjugate pair when projecting the matrices. I think we should just test some examples where this case occurs (e.g. the example @pmli mentioned) and then see if we get good results if we force the order of the ROM to be exactly `r`.

Reply:
I was curious about what sssMOR does. It looks like it just [warns](https://github.com/MORLab/sssMOR/blob/2f90150e138d0ec9e9c8cfc616c9aac250fcac7b/src/MOR/classic/modalMor.m#L244-L262) when the complex poles do not come in pairs and returns a complex ROM... I guess there could be situations when it's not possible to do anything else (e.g., if the FOM only has complex poles and `r` is odd), but maybe it's possible to do something otherwise (it reminds me of the knapsack problem...).

Reply:
So I tested the example above and compared the relative h2 error.
At first I reduced the model to order 2 with the current implementation that returns a real reduced model:
`rel. error for samdp 0.344 ,eig 0.430`
second I reduced the model to order 2 but now taking the complex eigenvector of the chosen pole from the pair: 
`rel. error for samdp 0.563, eig 0.615`
third I reduced the model to order 3:
` rel. error for samdp 0.144, eig 0.203`

From this first impression my suggestion is to take the current implementation which results in a real reduced model and would take just one pole of the complex pair but then warn the user such that the user needs to decide if he increases the reduced order by one to avoid the splitting of the pair. But I would try to test a bit more with larger problems. Do you have more suggestions which model could be good for testing?

Reply:
> But I would try to test a bit more with larger problems. Do you have more suggestions which model could be good for testing?

I would suggest testing with the [Slycot benchmarks](http://slicot.org/20-site/126-benchmark-examples-for-model-reduction). You can get the `LTIModel` from the .mat files with the `LTIModel.from_mat_file` function. Many of the models there have complex poles, e.g. the first few dominant poles (computed with samdp and which='LR') of the FOM model are complex conjugate pairs. I would suggest doing some tests with that model and some low orders for the ROM such as 3, 4, 5.